pkg/util/iptables/testing/parse_test.go

```diff
@@ -0,0 +1,552 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/lithammer/dedent"
+
+	"k8s.io/kubernetes/pkg/util/iptables"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+func mustParseRule(rule string) *Rule {
+	parsed, err := ParseRule(rule, false)
```

Comment:
Using the function under test in the stest is kind of cyclical, innit?  If there's a bug in ParseRule, the test will pass becuase the expected result will have the same bug...

Reply:
If you put TestParseRule first, and comment here why this is OK, I am happy